lib/active_merchant/billing/gateways/merchant_warrior.rb

```diff
@@ -142,6 +144,12 @@ def add_amount(post, money, options)
         post['hash'] = verification_hash(amount(money), currency)
       end
 
+      def add_recurring_flag(post, options)
+        return if options[:recurring_flag].nil?
+
+        post['recurringFlag'] = options[:recurring_flag]
```

Comment:
Very minor suggestion to consider:

```
  post['recurringFlag'] = options[:recurring_flag] if options[:recurring_flag]
```

Reply:
Does this provide anything that the guard clause on line 148 doesn't do?

Reply:
Nope! It's the same thing just written slightly differently. Either way works. If you prefer the guard clause approach, that's more than fine, although I might tweak it a bit to
```
return unless options[:recurring_flag]
```

Reply:
Got it, thanks! I think I will leave as is so that it has the option of being a boolean flag in the future if needed.